### PR TITLE
Gregg Heller Lesson 4

### PR DESCRIPTION
**Pull request by GreggHeller1**

### CODE REVIEW

**Comment by GreggHeller1** [comment]
Also now includes lesson 5

File: students/Gregg/lesson4/mailroom.py [hunk 1]
```diff
@@ -1,205 +1,259 @@
...
-        return False
+
+def primary_menu():
+    """Guide user through the mailroom UI"""
+    menu(primary_prompt, primary_switch_dict, quit_string='Q')
+
+
+primary_prompt = (
+    "\nWould you like to Send a Thank You (enter TY), "
+    "Create a Report (enter CR), send thank you letters to everyon (enter SL) or quit (enter Q).\n"
```
Comment (charlesrobison) [inline students/Gregg/lesson4/mailroom.py]: Typo - "everyone"

File: students/Gregg/lesson4/mailroom.py [hunk 2]
```diff
@@ -222,12 +276,5 @@ def tests():
...
-    return_to_primary_prompt = [False]
-    while(True):
-        return_to_primary_prompt[0] = False
-        action = input(primary_prompt())
-        if action == "Q":
-            break
-        else:
-            handled = handle_primary_prompt(action)
+    print("Welcome to Gregg's Mailroom 1.1\n")
+    primary_menu()
```
Comment (charlesrobison) [inline students/Gregg/lesson4/mailroom.py]: This is working well - nice job!

File: students/Gregg/lesson4/trigram.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                if len(next_options) == 0:
+                    return None
+        else:
+            return final_word_list
+
+
+
+if __name__ == "__main__":
+    trigram_dict = trigram_dict(short_path)
+    print(trigram_text(trigram_dict, 300))
```
Comment (charlesrobison) [inline students/Gregg/lesson4/trigram.py]: Looks good!

File: students/Gregg/lesson5/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+    The columns are name, total, number and average
+    """
+    row_list = []
+    headers = 'Donor name', 'Total Given', 'Num Gifts', 'Average Gift'
+    format_column_header = "{:<26}|{:^13}|{:^13}|{:^13}".format
+    row_list.append(format_column_header(*headers))
+    row_list.append('_' * 68)
+
+    donor_rows_list = [get_row(donor) for donor in donors]
```
Comment (charlesrobison) [inline students/Gregg/lesson5/mailroom.py]: Nice

File: students/Gregg/lesson5/mailroom.py [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+def tests():
+    """Test the functions"""
+    pass
+
+
+if __name__ == "__main__":
+    tests()
+    print("Welcome to Gregg's Mailroom 1.3\n")
+    primary_menu()
```
Comment (charlesrobison) [inline students/Gregg/lesson5/mailroom.py]: Looks good - nice use of exceptions and getting a comprehension in there.  Same suggestions for Lesson 4 mailroom apply here.  Let me know if you have any questions.

**Pull request merged**